crates/astria-sequencer/src/api_state_ext.rs

```diff
@@ -375,3 +375,350 @@ pub(crate) trait StateWriteExt: StateWrite {
 }
 
 impl<T: StateWrite> StateWriteExt for T {}
+
+#[cfg(test)]
+mod test {
+
+    use std::collections::HashMap;
+
+    use astria_core::{
+        sequencer::v1::{
+            asset::Id,
+            block::{
+                Deposit,
+                SequencerBlock,
+            },
+            Address,
+            RollupId,
+        },
+        Protobuf,
+    };
+    use cnidarium::StateDelta;
+    use rand::Rng;
+    use sha2::{
+        Digest as _,
+        Sha256,
+    };
+    use tendermint::{
+        account,
+        block::{
+            header::Version,
+            Header,
+        },
+        AppHash,
+        Hash,
+        Time,
+    };
+
+    use super::*;
+    use crate::proposal::commitment::generate_rollup_datas_commitment;
+
+    // creates new sequencer block, optionally shifting all values except the height by 1
+    fn make_test_sequencer_block(height: u32) -> SequencerBlock {
+        let mut rng = rand::thread_rng();
+
+        // create cometbft header
+        let mut header = Header {
+            app_hash: AppHash::default(),
+            chain_id: "test".to_string().try_into().unwrap(),
+            consensus_hash: Hash::default(),
+            data_hash: Some(Hash::default()),
+            evidence_hash: None,
+            height: height.into(),
+            last_block_id: None,
+            last_commit_hash: None,
+            last_results_hash: None,
+            next_validators_hash: Hash::default(),
+            proposer_address: account::Id::try_from([0u8; 20].to_vec()).unwrap(),
+            time: Time::now(),
+            validators_hash: Hash::default(),
+            version: Version {
+                app: 0,
+                block: 0,
+            },
+        };
+
+        // create inner rollup id/tx data
+        let mut deposits = HashMap::new();
+        for _ in 0..2 {
+            let rollup_id = RollupId::new([rng.gen(); 32]);
```

Comment:
just curious, can you do:
```suggestion
            let rollup_id = RollupId::new(rng.gen());
```

Reply:
yes you can! changed